core/src/main/scala/org/apache/spark/sql/rapids/tool/ToolUtils.scala

```diff
@@ -43,18 +46,59 @@ object ToolUtils extends Logging {
     org.apache.spark.SPARK_VERSION
   }
 
-  lazy val getEventFromJsonMethod: (String) => org.apache.spark.scheduler.SparkListenerEvent = {
+  lazy val getEventFromJsonMethod:
+    (String) => Option[org.apache.spark.scheduler.SparkListenerEvent] = {
     // Spark 3.4 and Databricks changed the signature on sparkEventFromJson
+    // Note that it is preferred we use reflection rather than checking Spark-runtime
+    // because some vendors may back-port features.
     val c = Class.forName("org.apache.spark.util.JsonProtocol")
-    // Explicitly check against the spark version.
-    if (isSpark340OrLater()) {
-      val m = c.getDeclaredMethod("sparkEventFromJson", classOf[String])
-      (line: String) =>
-        m.invoke(null, line).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
-    } else {
-      val m = c.getDeclaredMethod("sparkEventFromJson", classOf[org.json4s.JValue])
-      (line: String) =>
-        m.invoke(null, parse(line)).asInstanceOf[org.apache.spark.scheduler.SparkListenerEvent]
+    val m = Try {
+      // versions prior to spark3.4
```

Comment:
Thanks for adding reflection for checking the className. I hadn't considered the case where vendors may back-port features.